src/unix/fs.c

```diff
@@ -356,8 +359,11 @@ static int uv__fs_mkstemp(uv_fs_t* req) {
     r = -1;
   }
 
-  if (req->cb != NULL)
+  if (req->cb != NULL) {
+    /* Tell analyzer we earlier acquired the read lock */
+    uv_rwlock_assume_rdlocked(&req->loop->cloexec_lock);
     uv_rwlock_rdunlock(&req->loop->cloexec_lock);
```

Comment:
For my edification, this is necessary because clang can't figure out that locking and unlocking is guarded on the same `req->cb != NULL` condition? Does it help when it's written as e.g.:
```c
const bool has_callback = (req->cb != NULL);
if (has_callback)
  uv_rwlock_rdlock(&req->loop->cloexec_lock);
// ...
if (has_callback)
  uv_rwlock_rdunlock(&req->loop->cloexec_lock);
```

Reply:
Note that internally the analysis does similar trickery to make trylock work, pretending the lock isn't actually acquired until a branch on the result (so it would also fail to notice if there was a `!require`/`exclude` function called in between the trylock and the branch): https://clang.llvm.org/docs/ThreadSafetyAnalysis.html#try-acquire-bool-try-acquire-shared-bool